protocol/policy.go

```diff
@@ -9,34 +9,38 @@ import (
 type Timestamp uint64
 
 type Policies struct {
-	LibVersion    string
-	HashID        string
-	vrfPrivateKey vrf.PrivateKey
 	VrfPublicKey  vrf.PublicKey
 	EpochDeadline Timestamp
+	LibVersion    string
+	HashID        string
 }
 
-func NewPolicies(epDeadline Timestamp, vrfPrivKey vrf.PrivateKey) *Policies {
-	vrfPublicKey, ok := vrfPrivKey.Public()
-	if !ok {
-		panic(vrf.ErrorGetPubKey)
-	}
+func NewPolicies(epDeadline Timestamp, vrfPublicKey vrf.PublicKey) *Policies {
 	return &Policies{
-		LibVersion:    Version,
-		HashID:        crypto.HashID,
-		vrfPrivateKey: vrfPrivKey,
 		VrfPublicKey:  vrfPublicKey,
 		EpochDeadline: epDeadline,
+		LibVersion:    Version,
+		HashID:        crypto.HashID,
 	}
 }
 
 // Serialize encodes the policy to a byte array with the following format:
 // [lib version, cryptographic algorithm in use, epoch deadline, vrf public key]
 func (p *Policies) Serialize() []byte {
 	var bs []byte
+	bs = append(bs, p.VrfPublicKey...)                             // vrf public key
+	bs = append(bs, util.ULongToBytes(uint64(p.EpochDeadline))...) // epoch deadline
 	bs = append(bs, []byte(p.LibVersion)...)                       // lib Version
 	bs = append(bs, []byte(p.HashID)...)                           // cryptographic algorithms in use
-	bs = append(bs, util.ULongToBytes(uint64(p.EpochDeadline))...) // epoch deadline
-	bs = append(bs, p.VrfPublicKey...)                             // vrf public key
 	return bs
 }
+
+// FIXME: I'm just being lazy :(
+func ParsePubKey(policies []byte) vrf.PublicKey {
+	return vrf.PublicKey(policies[0:vrf.PublicKeySize])
```

Comment:
Yup, that's "lazy" part in the FIXME there.

If we choose to go this route, we'd probably want to use a scheme of 1 byte for len, followed by len bytes, when serializing.